src/java/org/apache/cassandra/index/sasi/SASIIndex.java

```diff
@@ -113,8 +112,7 @@ public SASIIndex(ColumnFamilyStore baseCfs, IndexMetadata config)
         Tracker tracker = baseCfs.getTracker();
         tracker.subscribe(this);
 
-        SortedMap<SSTableReader, Map<ColumnMetadata, ColumnIndex>> toRebuild = new TreeMap<>((a, b)
-                                                -> Integer.compare(a.descriptor.generation, b.descriptor.generation));
+        SortedMap<SSTableReader, Map<ColumnMetadata, ColumnIndex>> toRebuild = new TreeMap<>(Comparator.comparing(t -> t.descriptor.generation, SSTableUniqueIdentifierFactory.COMPARATOR));
```

Comment:
We can use `SSTableReader.generationComparator`:
```suggestion
        SortedMap<SSTableReader, Map<ColumnMetadata, ColumnIndex>> toRebuild = new TreeMap<>(SSTableReader.generationComparator);
```